### PR TITLE
[dart][dart-dio] More enum fixes for inner types

### PR DESCRIPTION
**Pull request by kuhnroyal**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

CC @swipesight (2018/09) @jaumard (2018/09) @josh-burton (2019/12) @amondnet (2019/12) @sbu-WBT (2020/12) @kuhnroyal (2020/12) @agilob (2020/12)

### CODE REVIEW

File: samples/openapi3/client/petstore/dart2/petstore_client_lib_fake/lib/model/enum_arrays.dart [hunk 1]
```diff
@@ -159,7 +159,7 @@ class EnumArraysArrayEnumEnum {
   const EnumArraysArrayEnumEnum._(this.value);
 
   /// The underlying value of this enum member.
-  final List<String> value;
```
Comment (kuhnroyal) [inline samples/openapi3/client/petstore/dart2/petstore_client_lib_fake/lib/model/enum_arrays.dart:162]: @agilob This commit also fixes the wrong value type that was appearrent in the other PR.

File: modules/openapi-generator/src/main/resources/dart2/class.mustache [hunk 2]
```diff
@@ -15,23 +15,8 @@ class {{{classname}}} {
...
-  Map<String, dynamic> {{{name}}};
-      {{/isMap}}
-    {{/isContainer}}
-    {{^isContainer}}
-      {{#allowableValues}}
-  {{{classname}}}{{{enumName}}} {{{name}}}{{#required}}{{#defaultValue}} = {{{classname}}}{{{enumName}}}._({{{defaultValue}}}){{/defaultValue}}{{/required}};
-      {{/allowableValues}}
-    {{/isContainer}}
   {{/isEnum}}
+  {{{datatypeWithEnum}}} {{{name}}};
```
Comment (kuhnroyal) [inline modules/openapi-generator/src/main/resources/dart2/class.mustache:19]: I am not sure about all of this change, especially the enum default value. Currently all default values are set in the constructor. As far as I can tell in that case a wrong constructor default value would be generated.

**Comment by agilob** [comment]
Took me long time to understand from class name whether this is correct or not... Do we **need** `...EnumEnum`? Do we need to append every enum class with `Enum`? Kind of excessive.

dart.dev doesn't recommend it https://dart.dev/guides/language/language-tour#enumerated-types 

No other generator does it. It was even difficult to reverse engineer to find original spec from the names.

File: modules/openapi-generator/src/main/resources/dart2/class.mustache (hunk 2 above)
Comment (agilob) [reply modules/openapi-generator/src/main/resources/dart2/class.mustache:19]: I think the default value should stay as null, unless field is marked as non-nullable, required or has default value provided, this looks good to me.

**Comment by kuhnroyal** [comment]
The `EnumEnum` is just happening because the model/property name already ends with `Enum`.
Appending `Enum` is done in `DefaultCodegen` and some generators do their own modifications but it is that way in several languages: https://github.com/OpenAPITools/openapi-generator/blob/master/samples/openapi3/client/petstore/java/native/src/main/java/org/openapitools/client/model/EnumArrays.java

This is already this way in the dart generator and has always been. (hence no changes to dart output, only dart-dio)

I am not particularly fond of the naming, I was actually thinking about removing it. Removing it will be a breaking change in the dart generator instead of dart-dio. Would be good to get some some more opinions on this.

**Review by agilob — APPROVED** [review]

**Pull request merged**